opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -512,7 +512,7 @@ def get_current_span(self) -> "Span":
     def start_span(
         self,
         name: str,
-        parent: ParentSpan = CURRENT_SPAN,
+        parent: Parent = CURRENT_SPAN,
```

Comment:
should these not be set to a SpanContext?

Reply:
I sure hope this parent is passed through to the sampler. And the sampler could very well use full parent span information (for example, I could have a sampler that discards any children of database spans (spans with the `db.instance` attribute set according to current semconv), as the details of a DB operation are often not interesting.

Reply:
OK, while my argument IMHO makes sense, this would currently actually violate the spec (https://github.com/open-telemetry/opentelemetry-specification/blob/c04041e16c9571b7ee5cfd5f102fc15e94ffacd7/specification/trace/sdk.md#shouldsample). I think this could use a spec PR. 😃

Reply:
Thanks for the link, I agree that this should be changed in the spec before we make a change here.

Reply:
to be clear: this is just renaming "ParentSpan" to "Parent", with the same type (Union[Span, SpanContext]). I would argue the previous name is a misnomer, since the argument can also accept a span.

I can rename it back though, if people feel otherwise.